### DISCUSSION TITLE
The problem of running CLI with complex output to the console and with their own interception of console output

### DISCUSSION THREAD

**Discussion by nsentinel** [nuke-build/nuke#1328] [Q&A]
### Usage Information

NUKE 8.0.0 / .NET 8.0 / Windows

### Description

The described problem is not really a bug, rather I would like to get advice on how to proceed in this situation.

We use NUKE among other things to run specialized internal utilities from source code (dotnet run) to prepare additional files. One of these utilities runs for quite a long time (a full cycle can take 20-30 minutes) and therefore outputs various progress bars in the process (using https://github.com/Mpdreamz/shellprogressbar), in addition, during the process various additional utilities are launched, the console output of which is captured.

In the current version of NUKE it is very badly combined, because NUKE tries to intercept and suppress part of the output, especially color output. Is there any way to completely disallow interception of console output when running a utility through NUKE?

### Reproduction Steps

```csharp
DotNetRun(_ => _
    .SetConfiguration(Configuration)
    .SetProjectFile(PrepareProject)
    .SetApplicationArguments($"--output {PrepareDirectory}")
);
```

### Expected Behavior

Full support for console progress bars and launching external utilities from code

### Actual Behavior

Suppression of progress bars and problems of getting redirected output

### Regression?

No

### Known Workarounds

No one that I'm aware of

### Could you help with a pull-request?

No

**Comment by A9G-Data-Droid** [comment]
Nuke is designed to run in a build pipeline. Most build pipelines show a log window, not a live terminal. So progress bars generally don't go well in a pipeline.

So you would need to move to a progress bar supported by your CI\CD platform. What are you using for CI\CD?

GitHub: https://github.com/orgs/community/discussions/160739
GitLab: https://gitlab.com/gitlab-org/gitlab/-/issues/371317
